numpyro/distributions/continuous.py

```diff
@@ -299,14 +299,18 @@ class StudentT(Distribution):
     reparametrized_params = ['loc', 'scale']
 
     def __init__(self, df, loc=0., scale=1., validate_args=None):
-        self.df, self.loc, self.scale = promote_shapes(df, loc, scale)
+        batch_shape = lax.broadcast_shapes(np.shape(df), np.shape(loc), np.shape(scale))
+        self.df = np.broadcast_to(df, batch_shape)
```

Comment:
Do we need to do this explicit `broadcast_to`?

Reply:
yes, it is required to have valid `chi2` samples (otherwise, the `chi2` samples will have a smaller shape than StudentT batch_shape

Reply:
I see. I was wondering if we can `broadcast_to` only for `df` but use `promote_shapes` for `loc`, `scale`, but let us profile that separately and make a change, if needed.

Reply:
yup, we don't need to broadcasting `loc`, `scale`. I'll fix that in LKJ pr. :)